action.yml

```diff
@@ -5,6 +5,15 @@ inputs:
     description: 'GitHub token'
     required: false
     default: ${{ github.token }}
```

Comment:
Can you check the code please to see what it's going to do if both are set? The default here means that `GITHUB_TOKEN` is always going to have a value.

I think we should make sure it prefers the app, probably? What do you think?

Reply:
I did checked GITHUB_TOKEN always won if set. Here I have to unset GITHUB_TOKEN to make it working https://github.com/grafana/k6-infrastructure/blob/deployment_tool_CI/.github/workflows/deployment-tools-cd.yaml#L172

Agree that it's better to prefer App. I can make a fix.

Reply:
Here is a change and test
https://github.com/grafana/wait-for-github/pull/72/commits/a2e5093d90c587aafa2ec70446f9d3e37cc553b4

Reply:
Can we move that one to a separate PR so that we can review it more carefully please? I think we should be good to go with your original one if you set it to an empty string.

Reply:
All right, I've moved the last commit to another branch.

> I think we should be good to go with your original one if you set it to an empty string.

I’m not sure I get the whole picture of how to do it right.
To set the default token input to an empty string, right? Well, this probably makes the logic of setting params a bit clear. Though in my use case, I have GITHUB_TOKEN env var setup for the workflow (https://github.com/grafana/k6-infrastructure/blob/deployment_tool_CI/.github/workflows/deployment-tools-cd.yaml#L21), so I must unset it anyway (https://github.com/grafana/k6-infrastructure/blob/deployment_tool_CI/.github/workflows/deployment-tools-cd.yaml#L172).
In GitHub actions, if a user provides the environment variable directly in the workflow step, then the value they set will take precedence over the value set by your action’s env field in action.yaml.

Generally I now think that this way:
```
 env:
     GITHUB_TOKEN: ${{ inputs.token }}
     GITHUB_CI_CHECKS: ${{ inputs.checks-to-wait-for }}
     GITHUB_APP_ID: ${{ inputs.app-id }}
    ...
   args:
     - --log-level
     - debug
    - --timeout
    - ${{ inputs.timeout }}
    - --recheck-interval
    - ${{ inputs.interval }}
  ...
    ```
    when some inputs go as args other as env vars is error-prone, env vars can be overwritten in the workflow step, so in the workflow, the env var can take priority over the argument. (At least for me, the natural order is that arg must have precedence, and env var is a fallback). So it probably should be refactored as all inputs go via args.
Having said that, I don’t know how this action is used in reality; I’m talking only from a single use-case perspective.

Reply:
You don't have to think about how the arguments are passed when you use the action, do you? It's all just GitHub inputs at that point.

The main reason for the split is for when the thing is running, though. Sensitive information like keys should not be put in command-line arguments because then it shows up in the process list and everyone who can access the system can see them.

I think we can merge this PR now that you have removed that change. For now, setting the input to an empty string explicitly should work to make the program use the app.

And then please submit that change again as a separate PR and we can chat about it on there, if there's any feedback for me to give.

Cheers :+1: